### DISCUSSION TITLE
Server timeouts after X time of not connecting.

### DISCUSSION THREAD

**Discussion by alexaptib** [coder/code-server#4081] [Q&A]
I have a program running on my node, and at night it always shuts down or whenever I am not connected through my browser. Is there any way to have it always up?

**Comment by bpmct** [comment]
Are you using `code-server --link`? Is the entire node shutting down or just code-server? 

Do you see any logs from code-server?

**Reply by jsjoeio** [reply]
Where do you have it deployed/running?

It may help (for my sake) to share an example. Let's say I have code-server installed on my iMac. I can run `code-server` to start it. As long as my mac stays on (and awake?), code-server should never stop. 

I'm guessing services like Digital Ocean, Heroku, etc. shut down if no use is detected. cc @bpmct who probably knows more

**Reply by bpmct** [reply]
Ah I see. What happens when you attempt to visit it again?

**Reply by bpmct** [reply]
After it has timed out?

**Reply by bpmct** [reply]
ah so it has to do with the program running in the terminal?

**Reply by jsjoeio** [reply]
Hmm... I don't think code-server _should_ shut down on it's own. I tried looking but couldn't find any docs for Umbrel. 

cc @code-asher

**Reply by lg-nineoneninethreenine** [reply]
Wait what if it is the problem with the Raspberry Pi? As far as I know, servers run as long as they are awake, so it might be that the Pi 3 has some sleep mechanism that activates at night and deactivates at morning. But is that really possible?

**Reply by lg-nineoneninethreenine** [reply]
Wait does this mean there is no downtime when code-server is not running(even with systemctl command)?

**Reply by lg-nineoneninethreenine** [reply]
So there is some sleep mechanism in code-server, is there?

**Reply by lg-nineoneninethreenine** [reply]
It has to be highly unlikely, I guess. Have you tried `sudo systemctl enable code-server@$USER`? It is a background process by the way. By the way @bpmct seems to have a Pi 4, has he noticed the error anywhere?

**Reply by code-asher** [reply]
I think terminals are shut down once the connection is closed after a short period of time. Something like 60 seconds. We could patch VS Code to persist them indefinitely.

**Reply by jsjoeio** [reply]
Opened an issue to track this: https://github.com/cdr/code-server/issues/4086
